### PR TITLE
Configurable toolbar

### PR DESCRIPTION
**Pull request by tiberiuichim**
I've refactored the Toolbar to allow defining it as configuration. I've tried not to go too crazy with the refactoring, although some of the components have very strong coupling (the More, the Contents dropdown and the rest of the "asides"). Probably it would be a good idea to further refactor that implementation, but for my purposes, to have the toolbar as configuration is enough.

### CODE REVIEW

**Review by ksuess — CHANGES_REQUESTED** [review]
Having a pluggable toolbar is a great progress. Some comments:

Great, very useful feature: header action of dropdown menu. 

I am missing a restriction to a special route as AppExtras configuration offers. See the following example where I want a button only for the /documentation section.
This could by now be achieved via a condition in custom button component. I think routes stuff should not go in component code.

```
export const settings = {
  ...defaultSettings,
  appExtras: [
    ...defaultSettings.appExtras,
    {
      match: '/documentation',
      component: AddBookmarkButton,
      props: {},
    },
  ],
};
```

File: docs/source/recipes/toolbar.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+  config.toolbar.activities.myCustomActivity = activity;
+}
+```
+
+Then the buttons would be used for a toolbar created like:
+
+```
+<Toolbar activity="myCustomActivity" inner={<></>} />
+```
```
[the remark this answers is not in the document]
Comment (tiberiuichim) [reply docs/source/recipes/toolbar.md:102]: Great feedback! Thank you! I'll adjust accordingly. The route idea is very good, indeed, I'll try to get it working. Feel free to push changes to this PR as well, if you have ideas, I really don't mind.

[the remark this answers is not in the document]
Comment (ksuess) [reply docs/source/recipes/toolbar.md]: > Great feedback! Thank you! I'll adjust accordingly. The route idea is very good, indeed, I'll try to get it working. Feel free to push changes to this PR as well, if you have ideas, I really don't mind.

may i mention this.

**Comment by ksuess** [comment]
Please stop hiding female work:

> tiberiuichim yesterday Author Member
> 
> Great feedback! Thank you! I'll adjust accordingly. The route idea is very good, indeed, I'll try to get it working. Feel free to push changes to this PR as well, if you have ideas, I really don't mind.
>

**Comment by tiberiuichim** [comment]
@ksuess: I'm sorry, that was not my intention. For the bits that's you've provided as review yesterday, Github provided a button "accept suggestions", then everything was automated. I don't even know how you've managed to get that diff showing in the Github UI.

**Comment by ksuess** [comment]
Hi @tiberiuichim, I am interested in a restriction of a toolbar menu/button to routes. Is it OK, if I have look at it, or did you already start to work on it?

**Comment by tiberiuichim** [comment]
@ksuess I'm sorry, I haven't had the time. Sure, you can go ahead with the required changes, either directly push to this branch or as a separate PR to this branch. Many thanks!

**Comment by tiberiuichim** [comment]
There's issues with the current implementation, some things I've fixed in https://github.com/plone/volto/commit/0ff4f18d16c6f2ed10b3d7451e765362c412e472

I've also noticed that I can't close the "More" menu by clicking on the document body.

**Comment by tiberiuichim** [comment]
Work continued as part of the Slots/Quanta PR

**Pull request closed without merging**